### ISSUE TITLE
RowVersionAttribute can't be used when deriving from ContainSagaData

### ISSUE THREAD

**Issue by dvdstelt** [Particular/NServiceBus#3776]
When using the `RowVersion` attribute in a saga data class, NServiceBus checks if it's on the aggregate root. Because subclasses should never have this attribute.

However when the new `ContainSagaData` class is used, NServiceBus thinks the root class is `ContainSagaData` and doesn't allow your (root) saga data class to contain the attribute. It should be checked if `ContainSagaData` is the immediate base class for the saga data class and allow it anyway.

**Comment by danielmarbach** [comment]
I think I submitted this base class in the past as an external contributor to the core. Reflecting about it again I think it is pretty useless to have a base class for an interface with three properties. I would be Ok to deprecate it.

**Comment by timbussmann** [comment]
This is only an issue when using NHibernate, right? Would implementing `IContainSagaData` instead allow to use the RowVersion attribute be an usable workaround?

I think the `ContainSagaData` class should remain, it removes noise for most users.

**Comment by dvdstelt** [comment]
Should we have rowversion automatically added as well, but just in NH persister? That would definitely add something, I think. Not sure on solution to choose though.

**Comment by timbussmann** [comment]
> Should we have rowversion automatically added as well, but just in NH persister?

do all nhibernate users want/need to use that attribute?

**Comment by danielmarbach** [comment]
I originally proposed this base class. In the meantime I think it is gold
plating of DRY. A bade class for 3 properties? Really?

**Comment by timbussmann** [comment]
three properties regular users probably don't care about. So adding them for every saga data seems to be unnecessary noise? I don't we should remove if it's only for nhibernate users (even all of them?) in case we have a relative simple workaround?

**Comment by andreasohlund** [comment]
The saga master class people seems to like it since they wouldn't have the "infastructure stuff" that we need right in their face all the time.

Side note: This becomes a non issue when we move to the new saga model - https://github.com/Particular/PlatformDevelopment/issues/546 since that would move all that cruft out of the users way.

I think it does add value so I'm :+1:  to keep it

**Comment by andreasohlund** [comment]
Ok to close this?

**Comment by dvdstelt** [comment]
I was working on a PR to fix this, just before my vacation. It's probably only a small change which makes this work again, but I got stuck with some stuff in NHibernate. I wanted to contact @SzymonPobiega , but since he was on vacation at the time, I delayed it. I'd like to have a look at it with him and see if it can be fixed quickly. If it's too much work, I'll report the findings here and close it. Is that ok?
